lib/matplotlib/cbook/__init__.py

```diff
@@ -1914,20 +1914,28 @@ class so far, an alias named ``get_alias`` will be defined; the same will
     if cls is None:  # Return the actual class decorator.
         return functools.partial(_define_aliases, alias_d)
 
-    def make_alias(name):  # Enforce a closure over *name*.
-        def method(self, *args, **kwargs):
```

Comment:
Just add a `@functools.wraps(getattr(cls, name))` decorator here?  inspect.signature will then get the signature from the wrappee.  Note that you can even play with the `assigned` kwarg to functools.wraps if you don't want to update the name...

Reply:
Thanks for the hint. That would work too.

The getattr-alias is a bit slower than my explictly coded lambda (67ns vs. 26ns). I mean, it's not much in absolute values, but the aliases might be used a lot by users. Do we want the added speedup or should we stay with the slightly simpler but slower getattr implementiation.

Reply:
Well, lambdas also don't have a `__name__`/`__qualname__` properly set, which doesn't help https://github.com/matplotlib/matplotlib/pull/11936 :-)

Reply:
No strong opinion there.

Reply:
There are 60 properties. The package import time does not measureably change between the two approaches (200.8 +/- 6.0ms vs. 201.0 +/- 6.0ms).

Probably both approaches are fast enough in startup as well as runtime use to not make a difference. I will go back to decorating the original function to keep it simpler. If runtime performance is really super-critical, users should just use the methods functions, not the aliases.